### PR TITLE
Display App Name in redirection page of SAML Application

### PR DESCRIPTION
**Pull request by kayathiri4**
**Purpose**
 In the authentication flow of the SAML application, the default Assertion consumer URL is getting displayed while redirecting back the user after successful authentication.
 This PR fixes by displaying the app name in the redirection page.

### CODE REVIEW

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java [hunk 1]
```diff
@@ -1050,10 +1050,29 @@ private void generateSamlPostPageFromFile(HttpServletResponse resp, String acUrl
...
+        String tenantDomain;
         String finalPage;
+
+        issuer = SAMLSSOUtil.getIssuerWithQualifierInThreadLocal();
+        tenantDomain = SAMLSSOUtil.getTenantDomainFromThreadLocal();
         String htmlPage = IdentitySAMLSSOServiceComponent.getSsoRedirectHtml();
-        String pageWithAcs = htmlPage.replace("$acUrl", acUrl);
-        String pageWithAcsResponse = pageWithAcs.replace("<!--$params-->",
+        String pageWithAcs = htmlPage.replace("$acUrl", acUrl);;
+        String pageWithApp = pageWithAcs.replace("$app", acUrl);;
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java:1061]:
```suggestion
        String pageWithAcs = htmlPage.replace("$acUrl", acUrl);
        String pageWithApp = pageWithAcs.replace("$app", acUrl);
```

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java [hunk 2]
```diff
@@ -1050,10 +1050,29 @@ private void generateSamlPostPageFromFile(HttpServletResponse resp, String acUrl
...
+
+        if (!issuer.isEmpty() && !tenantDomain.isEmpty()){
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java:1063]: Lets make sure these are null safe

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java [hunk 3]
```diff
@@ -1050,10 +1050,29 @@ private void generateSamlPostPageFromFile(HttpServletResponse resp, String acUrl
...
+            try {
+                String spName = ApplicationManagementService.getInstance()
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java:1065]: I think same as the acUrl the spName should also be passed into this method.

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java [hunk 4]
```diff
@@ -975,14 +975,27 @@ private void sendResponse(HttpServletRequest req, HttpServletResponse resp, Stri
...
         acUrl = getACSUrlWithTenantPartitioning(acUrl, tenantDomain);
+        String issuer = SAMLSSOUtil.getIssuerWithQualifierInThreadLocal();
 
         if (acUrl == null || acUrl.trim().length() == 0) {
             // if ACS is null. Send to error page
             log.error("ACS Url is Null");
             throw IdentityException.error("Unexpected error in sending message out");
         }
 
+        if (issuer != null && tenantDomain != null && !issuer.isEmpty() && !tenantDomain.isEmpty()){
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java:988]: lets use StringUtils.isNotBlank
```suggestion
        if (StringUtils.isNotBlank(issuer) && StringUtils.isNotBlank(tenantDomain)){
```

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java [hunk 5]
```diff
@@ -2143,4 +2159,26 @@ private String getLoggedInTenantDomain(HttpServletRequest req) {
...
+        String tenantDomain = SAMLSSOUtil.getTenantDomainFromThreadLocal();
+        String issuer = SAMLSSOUtil.getIssuerWithQualifierInThreadLocal();
+
+        if (StringUtils.isNotBlank(issuer) && StringUtils.isNotBlank(tenantDomain)){
+            try {
+                 return ApplicationManagementService.getInstance()
+                        .getServiceProviderNameByClientId(SAMLSSOUtil.splitAppendedTenantDomain(issuer),
+                                IdentityApplicationConstants.Authenticator.SAML2SSO.NAME, tenantDomain);
+            } catch (IdentityApplicationManagementException e) {
+                log.error("Error while getting Service provider name for issuer:" + issuer + " in tenant: " +
```
Comment (pamodaaw) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java:2178]:
```suggestion
                log.error("Error while getting service provider name for issuer:" + issuer + " in tenant: " +
```

**Review by pamodaaw — APPROVED** [review]

**Pull request merged**